modules/standard/Random.chpl

```diff
@@ -293,8 +337,21 @@ module Random {
         compilerError("NPBRandomStream(eltType=", eltType:string, 
                       ") can only be used to fill arrays of ", eltType:string);
       }
-        
-      pragma "no doc"
+
+      /*
+         Returns an iterable expression for generating
+         D.numIndices random numbers. The RNG state will
+         be immediately advanced by D.numIndices before
+         this function yields any values.
+
+         The iterable expression can be usefully used
+         in a parallel context, whether standalone or
+         zippered.
+
+         :arg D: a domain
+         :arg resultType: the type of number to yield
+         :return: an iterable expression yielding random resultType values
+       */
```

Comment:
It seems reasonable to me.
 * It does not need to return an iterator to match this interface (it could return a record with a .these() method)
 * It would be easy for us to make it take in a range too
 * I can't think of a better name, other than 'these', which we can't use for here without Leader/Follower 2.0

I think it's more unreasonable for us to leave it undocumented but use it in benchmarks/exercises/primers than it is to deprecate it in the future if we have something better. And the main thing that I know of that would be better is Leader/Follower 2.0. I think this is a reasonable interim solution and I am unphased by the deprecation procedure.

In fact, since this has been used in benchmarks/exercises/primers, I'd argue if we remove it we should *already* deprecate it. You might not remember this, but release/examples/primers/randomNumbers.chpl demonstrates this method: 


      //
      // The stream can be used to iterate over a specified set of positions.
      //
      for i in randStreamSeeded.iterate({5..10}, real) {
        writeln(i);
      }